lib/Xmds/Soap.php

```diff
@@ -1447,40 +1464,52 @@ protected function doSubmitStats($serverKey, $hardwareKey, $statXml)
             if ($tag == 'null')
                 $tag = null;
 
-            // Add this information to an array for batch insert
-            $stats[] = [
-                'type' => $type,
-                'statDate' => $now,
-                'fromDt' => $fromdt,
-                'toDt' => $todt,
-                'scheduleId' => $scheduleId,
-                'displayId' => $this->display->displayId,
-                'layoutId' => $layoutId,
-                'mediaId' => $mediaId,
-                'tag' => $tag,
-                'widgetId' => $widgetId,
-            ];
-        }
+            if (($type == 'media') || ($type == 'widget')) {
+                $mediaStats[] = [
+                    'type' => $type,
+                    'statDate' => $now,
+                    'fromDt' => $fromdt,
+                    'toDt' => $todt,
+                    'scheduleId' => $scheduleId,
+                    'displayId' => $this->display->displayId,
+                    'layoutId' => $layoutId,
+                    'mediaId' => $mediaId,
+                    'tag' => $tag,
+                    'widgetId' => $widgetId,
+                ];
+            } elseif ($type == 'layout') {
+                $layoutStats[] = [
+                    'type' => $type,
+                    'statDate' => $now,
+                    'fromDt' => $fromdt,
+                    'toDt' => $todt,
+                    'scheduleId' => $scheduleId,
+                    'displayId' => $this->display->displayId,
+                    'layoutId' => $layoutId,
+                ];
```

Comment:
The other `if` statement needed here is called `event` and records tag data.

Reply:
I added as below: when $type == 'event'
```
} elseif ($type == 'event') {
    $tagStats[] = [
        'type' => $type,
        'statDate' => $now,
        'fromDt' => $fromdt,
        'toDt' => $todt,
        'scheduleId' => $scheduleId,
        'displayId' => $this->display->displayId,
        'tag' => $tag,
    ];
}
```